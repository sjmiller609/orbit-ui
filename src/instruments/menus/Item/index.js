'use strict'
import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import s from './styles.scss'
import { Link } from 'instruments'

const Item = ({
  to,
  children,
  active,
  className,
  activeClassName,
  ...props
}) => {
  return (
    <Link
      {...props}
      to={to}
      exact={true}
      className={classnames(s.item, active && s.active, className)}
      activeClassName={classnames(s.active, activeClassName)}>
      {children}
    </Link>
  )
}

Item.propTypes = {
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  className: PropTypes.string,
  activeClassName: PropTypes.string,
  to: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  active: PropTypes.bool,
}

export default Item
