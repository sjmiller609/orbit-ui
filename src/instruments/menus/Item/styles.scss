@import 'src/styles/mixins.scss';

a.item {
  display: block;
  font-size: 1.4rem;
  color: $darkGray;

  &:hover,
  &:focus {
    color: $lightPurple;
    img {
      opacity: 1;
    }
  }

  img {
    opacity: 0.5;
  }
  // this prevents width from changing on hover
  &::after {
    display: block;
    content: attr(title);
    @include font($f700);
    height: 0;
    overflow: hidden;
    visibility: hidden;
  }
}

a.active {
  @include font($f700);
  color: $lightPurple;
}
