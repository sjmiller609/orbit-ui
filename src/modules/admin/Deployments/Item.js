'use strict'
import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import s from './styles.scss'
import { TableRow, Box, P, Mini, H3, ShowDate, Button } from 'instruments'

const Item = ({ deployment, className, handleClick }) => {
  const workspaceId = deployment.workspace.id
  const to =
    '/workspaces/' + workspaceId + '/deployments/' + deployment.releaseName
  const config = to + '/configure'

  const columns = [
    <Box key="0" align="flex-start" className={s.title}>
      <H3>{deployment.label}</H3>
      <P>{deployment.releaseName}</P>
    </Box>,
    <Box key="2" align="flex-start" className={s.log}>
      <P className={s.subTitle}>Deployed</P>
      <Mini>
        <ShowDate date={deployment.createdAt} />
      </Mini>
    </Box>,
    <Box key="3" align="flex-start" className={s.log}>
      <P className={s.subTitle}>Updated</P>
      <Mini>
        <ShowDate date={deployment.updatedAt} />
      </Mini>
    </Box>,
    <Box key="1" align="flex-start" className={s.log}>
      <P className={s.subTitle}>Version</P>
      <P>{deployment.version}</P>
    </Box>,
    <Box key="4" align="flex-start" className={s.upgrade}>
      <Button onClick={() => handleClick(config)} to={null}>
        Config
      </Button>
    </Box>,
  ]

  return (
    <TableRow className={classnames(s.item, className)} columns={columns} />
  )
}

Item.propTypes = {
  deployment: PropTypes.object,
  latestVersion: PropTypes.string,
  className: PropTypes.string,
  handleClick: PropTypes.func,
}

export default Item
