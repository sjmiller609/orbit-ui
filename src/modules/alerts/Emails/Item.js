'use strict'
import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import s from './styles.scss'
import { TableRow, Box, B } from 'instruments'

const Item = ({ email, className, ...props }) => {
  const columns = [
    <Box key="0" align="flex-start" className={s.title}>
      <B>{email}</B>
    </Box>,
  ]

  return (
    <TableRow
      {...props}
      className={classnames(s.item, className)}
      columns={columns}
    />
  )
}

Item.propTypes = {
  email: PropTypes.string,
  className: PropTypes.string,
}

export default Item
