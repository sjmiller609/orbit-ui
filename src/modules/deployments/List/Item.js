'use strict'
import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import s from './styles.scss'
import { TableRow } from '../../../instruments'

const Item = ({ deployment, className }) => {
  const columns = [
    <p key="1">{deployment.title}</p>,
    <p key="2">{deployment.release_name}</p>,
    <p key="3">logs</p>,
  ]

  const to = '/deployments/' + deployment.release_name

  return (
    <TableRow
      className={classnames(s.item, className)}
      columns={columns}
      to={to}
    />
  )
}

Item.propTypes = {
  deployment: PropTypes.object,
  className: PropTypes.string,
}

export default Item
