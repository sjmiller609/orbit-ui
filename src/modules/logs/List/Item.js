'use strict'
import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import s from './styles.scss'
import { TableRow, Box, P, ShowDate } from 'instruments'

const Item = ({ log, className }) => {
  const columns = [
    <Box key="0" align="flex-start" className={s.log}>
      <P>
        > <ShowDate date={log.date} /> {log.log}
      </P>
    </Box>,
  ]

  return (
    <TableRow className={classnames(s.item, className)} columns={columns} />
  )
}

Item.propTypes = {
  log: PropTypes.object,
  className: PropTypes.string,
}

export default Item
