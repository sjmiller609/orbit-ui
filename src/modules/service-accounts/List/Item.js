'use strict'
import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import s from './styles.scss'
import { TableRow, Box, P, Mini, H3, ShowDate, Icon } from 'instruments'

const Item = ({ serviceAccount, path, className }) => {
  const columns = [
    <Box key="0">
      <Icon icon="satellite" className={s.icon} />
    </Box>,
    <Box key="1" align="flex-start" className={s.title}>
      <H3>{serviceAccount.label}</H3>
      <P>{serviceAccount.category}</P>
    </Box>,
    <Box key="2" align="flex-start" className={s.log}>
      <P className={s.subTitle}>
        {serviceAccount.lastUsedAt ? 'Last Used' : 'Created At'}
      </P>
      <Mini>
        <ShowDate
          date={serviceAccount.lastUsedAt || serviceAccount.createdAt}
        />
      </Mini>
    </Box>,
  ]

  const to = path + '/' + serviceAccount.id

  return (
    <TableRow
      className={classnames(s.item, className)}
      columns={columns}
      to={to}
    />
  )
}

Item.propTypes = {
  serviceAccount: PropTypes.object,
  className: PropTypes.string,
  path: PropTypes.string,
}

export default Item
