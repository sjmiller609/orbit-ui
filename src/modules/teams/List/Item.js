'use strict'
import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import s from './styles.scss'
import {
  TableRow,
  Box,
  P,
  Mini,
  H3,
  ShowDate,
  Icon,
} from '../../../instruments'

const Item = ({ team, className }) => {
  const columns = [
    <Box key="0" className={s.icon}>
      T
    </Box>,
    <Box key="1" align="flex-start" className={s.title}>
      <H3>{team.label}</H3>
    </Box>,
    <Box key="2" align="flex-start" className={s.log}>
      <P className={s.subTitle}>Last updated</P>
      <Mini>
        <ShowDate date={team.updated_at} />
      </Mini>
    </Box>,
  ]

  const to = '/teams/' + team.id

  return (
    <TableRow
      className={classnames(s.item, className)}
      columns={columns}
      to={to}
    />
  )
}

Item.propTypes = {
  team: PropTypes.object,
  className: PropTypes.string,
}

export default Item
