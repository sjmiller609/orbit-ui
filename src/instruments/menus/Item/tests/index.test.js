import React from 'react'
import { shallow } from 'enzyme'
import Item from '../index'

describe('Item', () => {
  it('should be defined', () => {
    expect(Item).toBeDefined()
  })

  it('should render correctly', () => {
    const wrapper = shallow(<Item />)
    expect(wrapper).toMatchSnapshot()
  })
})
