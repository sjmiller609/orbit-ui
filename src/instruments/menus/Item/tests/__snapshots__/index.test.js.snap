// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should render correctly 1`] = `
<Link
  activeClassName="active"
  className="item"
  exact={true}
  title=""
/>
`;
